esp32/modesp.c

```diff
@@ -37,6 +38,47 @@
 #include "drivers/dht/dht.h"
 #include "modesp.h"
 
+#define MODESP_INCLUDE_CONSTANTS (1)
+
+STATIC mp_obj_t esp_osdebug(size_t n_args, const mp_obj_t *args) {
+    switch (n_args) {
+    case 1:
+        if (args[0] == mp_const_none) {
+            esp_log_level_set("*", ESP_LOG_ERROR);
+        } else {
+            esp_log_level_set("*", LOG_LOCAL_LEVEL);
+        }
+        break;
+
+    case 2:
+        esp_log_level_set("*", mp_obj_get_int(args[1]));
+        break;
+    }
+
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(esp_osdebug_obj, 1, 2, esp_osdebug);
```

Comment:
In the case of the ESP32 it doesn't really allow you to redirect errors to different UARTs (?) so I'd say it's fine to repurpose the single argument to this function to be the log level.